source/h1_connection.c

```diff
@@ -388,6 +388,34 @@ int aws_h1_stream_activate(struct aws_http_stream *stream) {
     return AWS_OP_SUCCESS;
 }
 
+void aws_h1_stream_cancel(struct aws_http_stream *stream, int error_code) {
```

Comment:
no-op: I was going to say "we should ensure they don't pass in 0", but turns it's fine. If they passed 0:
- the channel will end up shutting down with error_code 0
- BUT if the channel's error_code is 0, the stream will default to [AWS_ERROR_HTTP_CONNECTION_CLOSED](https://github.com/awslabs/aws-c-http/blob/0c965027a5747f427b3ac6b70e528be93ab21be4/source/h1_connection.c#L2081-L2082)